### DISCUSSION TITLE
threshold_maximum 0

### DISCUSSION THREAD

**Discussion by ceewanna** [Motion-Project/motionplus#34] [General]
In help, it reads "Maximum number of changed pixels that triggers an event. If the number of changed pixels is over the threshold maximum, no event is triggered. _A value of zero disables threshold_maximum_."

Based on my testing, it seems if threshold_maximum has to be set at a relatively high value (higher than threshold) for motion detection to get working. If threshold_maximum is left with 0, there is virtually no motion detection. I am not so sure if I get this correctly.

**Comment by Mr-Dave** [comment]
Upon initial view, this wouldn't seem right.  This section of the [code](https://github.com/Motion-Project/motionplus/blob/587491fbb7d759bf74a7cba7eb2a2f518f9d200a/src/motion_loop.cpp#L1363) sets the maximum used later in the criteria.  If the threshold_maximum specified by the user is greater than motion threshold, use the user provided threshold_maximum.  Otherwise set the threshold_maximum to the 1.5x the image size.
t

**Reply by ceewanna** [reply]
In my setting, while threshold and threshold_maximum are set in the main config, I don't have height and width set in there. They are however set individually for each camera feed. I am not sure if it has to do with this or not

Right from the beginning I had been wondering why it didn't detect any motion until one day I just changed threshold_maximum in my main config to just a few thousands and it worked.

**Comment by Mr-Dave** [comment]
I should listen better.  You are entirely correct and a fix has been pushed.
